pkg/util/oom/oom_linux.go

```diff
@@ -75,6 +75,7 @@ func applyOOMScoreAdj(pid int, oomScoreAdj int) error {
 			err = fmt.Errorf("failed to apply oom-score-adj to pid %d (%v)", pid, err)
 			continue
 		}
+		defer f.Close()
```

Comment:
It does - generally if you *must* guarantee the file is written, you need to do:

```
if _, err := f.Write(...); err != nil {
  f.Close()
  return err
}
if err := f.Close(); err != nil {
  return err
}
```

Defer ignores errors, and because Close() is actually when the filesystem guarantees the data is written, it's possible that your write (from above) is not actually persisted without returning an error.